scala-thoughts.md

```diff
@@ -0,0 +1,39 @@
+The first thing I would say about Scala is that there is too much syntax. It
+seems that there are several different ways of doing every simple thing, and
+which syntax you should use in each place is very unclear. For example, I'm
+still unclear what the difference between the following statements are:
+
+    f g x
+    f.g(x)
+    f(g(x))
+
```

Comment:
good point!

I think that `f g x` is executing the method `g` that is a part of the object `f` with the argument `x`
I think that the second line does that exact same thing, while I think that the last example would be the composition of two functions `f,g,` with the arguments `g(x),x` respectively
I was confused about this as well